File: csharp/SoapClient.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

public class SoapClient
{
    private readonly HttpClient _client;

    public SoapClient(string baseAddress)
    {
        _client = new HttpClient { BaseAddress = new Uri(baseAddress) };
    }

    private async Task SendRequestAsync(string action, int i)
    {
        try
        {
            var content = new StringContent(action, Encoding.UTF8, "text/xml");
            var response = await _client.PostAsync("", content);
            response.EnsureSuccessStatusCode();
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Request {i} failed: {ex.Message}");
        }
    }

    public async Task RunTests()
    {
        var soapActions = new[]
        {
            new
            {
                Name = "Usuarios",
                Action = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:spy=\"spyne.examples.flask\"><soapenv:Header/><soapenv:Body><spy:ler_usuarios/></soapenv:Body></soapenv:Envelope>"
            },
            new
            {
                Name = "Musicas",
                Action = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:spy=\"spyne.examples.flask\"><soapenv:Header/><soapenv:Body><spy:ler_musicas/></soapenv:Body></soapenv:Envelope>"
            },
            new
            {
                Name = "Playlists",
                Action = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:spy=\"spyne.examples.flask\"><soapenv:Header/><soapenv:Body><spy:ler_playlists/></soapenv:Body></soapenv:Envelope>"
            }
        };

        var numTests = new[] { 100, 200, 300 };

        foreach (var numTest in numTests)
        {
            Console.WriteLine($"{numTest} requests:");
            foreach (var soapAction in soapActions)
            {
                var start = DateTime.Now;

                for (int i = 0; i < numTest; i++)
                {
                    await SendRequestAsync(soapAction.Action, i);
                }

                var end = DateTime.Now;
                Console.WriteLine($"{soapAction.Name} : {(end - start).TotalSeconds} seconds");
            }
        }
    }
}

File: rust/src/soap_client.rs
use reqwest::Client;
use std::time::Instant;

const BASE_URL: &str = "http://localhost:5001/soap";

const SOAP_ACTIONS: &[(&str, &str)] = &[
    (
        "Usuarios",
        r#"<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:spy="spyne.examples.flask">
                       <soapenv:Header/>
                       <soapenv:Body>
                         <spy:ler_usuarios/>
                       </soapenv:Body>
                     </soapenv:Envelope>"#,
    ),
    (
        "Musicas",
        r#"<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:spy="spyne.examples.flask">
                      <soapenv:Header/>
                      <soapenv:Body>
                        <spy:ler_musicas/>
                      </soapenv:Body>
                    </soapenv:Envelope>"#,
    ),
    (
        "Playlists",
        r#"<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:spy="spyne.examples.flask">
                        <soapenv:Header/>
                        <soapenv:Body>
                          <spy:ler_playlists/>
                        </soapenv:Body>
                      </soapenv:Envelope>"#,
    ),
];

pub async fn run_tests() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new();
    let test_counts = vec![100, 200, 300];

    for &count in &test_counts {
        println!("{} requests:", count);
        for &(name, action) in SOAP_ACTIONS {
            let start = Instant::now();
            for _ in 0..count {
                let res = client
                    .post(BASE_URL)
                    .header("Content-Type", "text/xml")
                    .body(action)
                    .send()
                    .await;
                if let Err(err) = res {
                    eprintln!("Request failed: {:?}", err);
                }
            }
            let duration = start.elapsed();
            println!("{} : {:.6?} seconds", name, duration.as_secs_f64());
        }
    }

    Ok(())
}
